### PR TITLE
Add Poisson tutorial notebook and harden list1 case encoding

### PR DESCRIPTION
**Pull request by xywei**
## Summary
- add a full starfish Poisson tutorial notebook for `volumential` + `pytential` with clearer narrative, walkthrough sections, and richer diagnostics/visualizations for single-run, convergence, and AMR studies
- harden list1 interaction-case encoding for non-dyadic box extents via a configurable encoding bias and optional drift diagnostics to prevent case-id misclassification in OpenCL kernels
- update volume FMM/function-extension and regression tests to cover source-only tree handling, calculus-patch behavior, and related integration edge cases used in the notebook workflow

## Validation
- executed `examples/poisson2d_pytential_volumential.ipynb` end-to-end on `ipa` after updates
- verified uniform co-refinement trend (slope ~3.786) and AMR accuracy/cost reporting from notebook outputs

### CODE REVIEW

File: volumential/expansion_wrangler_fpnd.py [hunk 1]
```diff
@@ -651,6 +696,14 @@ def eval_direct_single_out_kernel(
             queue, self.tree, self.near_field_table[kname][0].n_q_points
         )
 
+        _validate_table_box_particle_layout(
+            queue,
+            self.tree,
+            target_boxes,
+            neighbor_source_boxes_lists,
```
[the remark this answers is not in the document]
Comment (xywei) [reply volumential/expansion_wrangler_fpnd.py]: Addressed in `6220327`: table-layout validation is now cached per wrangler (`_validate_table_box_particle_layout_cached`) and keyed by source/target box array identity plus q-order. This keeps fail-fast validation for new layouts but avoids repeated host readback/sync on every near-field call and across output kernels for the same traversal layout.

File: test/test_function_extension.py [hunk 2]
```diff
@@ -161,3 +165,215 @@ def test_harmonic_extension_to_volume_mesh_points_smoke(ctx_factory):
...
+    fx = unflatten(
+        bdry_nodes[0],
+        actx.from_numpy(np.ascontiguousarray(np.ones_like(xb))),
+        actx,
+    )
+    fy = unflatten(
+        bdry_nodes[0],
+        actx.from_numpy(np.ascontiguousarray(np.zeros_like(yb))),
+        actx,
+    )
```
[the remark this answers is not in the document]
Comment (xywei) [reply test/test_function_extension.py]: Fixed in `bce607d`: both biharmonic boundary fixtures now flatten coordinate leaves separately (`bdry_nodes[0]` and `bdry_nodes[1]`) before NumPy conversion, which keeps x/y vectors aligned with unflatten inputs.

File: test/test_volume_fmm.py [hunk 3]
```diff
@@ -49,6 +49,198 @@
...
+    levels = np.asarray(tob.box_levels, dtype=np.int32)
+    bad_edges_before = 0
+    for parent in range(tob.nboxes):
+        plevel = int(levels[parent])
+        for child in tob.box_child_ids[:, parent]:
+            child = int(child)
+            if child != 0 and int(levels[child]) != plevel + 1:
+                bad_edges_before += 1
+
+    assert bad_edges_before > 0
```
[the remark this answers is not in the document]
Comment (xywei) [reply test/test_volume_fmm.py]: Agreed, and fixed in `1031bdb`: the precondition check now skips when `bad_edges_before == 0` ("boxtree no longer reproduces the invalid-edge topology") instead of failing hard, so the test only asserts repair behavior when upstream still reproduces the broken topology.
